src/common/io.py

```diff
@@ -1607,9 +1612,10 @@ def get_variables_data(self,
             stop_index = len(time) + start_index
 
         for name in names:
-            trajectories.append(self._get_variable_data_as_trajectory(name, time, start_index, stop_index))
+            trajectories[name] = self._get_variable_data_as_trajectory(name, time, start_index, stop_index)
 
         new_start_index = start_index + len(time) if len(trajectories) > 0 else None
+        self._last_set_of_indices = (start_index, stop_index) # update them before we exit
```

Comment:
Should this not take into account that we've may already have read parts of the trajectory? I.e. this should not be the union of what is already read together with this new part?

Reply:
This is what I mean. However, we've read a trajectory so it should be easy to very that it it contains what has been asked, is it not?